heat/core/sanitation.py

```diff
@@ -27,6 +28,120 @@
 ]
 
 
+def sanitize_distribution(
+    *args: DNDarray, target: DNDarray, diff_map: torch.Tensor = None
+) -> Union[DNDarray, Tuple(DNDarray)]:
+    """
+    Make every arg have the same distribution (along split-axis) as the target,
+    such that after this sanitation, the lshapes are compatible wrt split-dimension.
+    Args can contain unsplit-DNDarrays.
```

Comment:
"Make every arg have the same distribution" -> "Distribute every ``arg`` according to ``target.lshape_map``"
"compatible wrt split-dimension" -> "compatible along the split dimension"
"unsplit-DNDarrays" -> "non-distributed ``DNDarrays``"

(TBC, need to check internal referencing within the docs)